ht/services.py

```diff
@@ -38,39 +38,126 @@ def image_result(item):
-			
+
 	return embed, image
-	
+
 async def ds_catalog(session, charge):
 	catalog = await utils.get_json(session, f"https://drawshield.net/api/catalog/{urllib.parse.quote(charge)}")
-	
+
 	if not catalog.startswith("http"): return None
 	return catalog.split("\n")
-	
+
 async def commons(session, loop, filename):
 	result_text = await utils.get_text(
 		session,
 		f"https://magnus-toolserver.toolforge.org/commonsapi.php?image={filename}&thumbwidth=600&thumbheight=600&meta"
 	)
 	get_json = lambda text_string, root: ElementTree.fromstring(text_string).find(root)
-	
-	return await loop.run_in_executor(None, get_json, result_text, "file")
\ No newline at end of file
+
+	return await loop.run_in_executor(None, get_json, result_text, "file")
+
+def is_option_keyword(s):
+	return s.startswith(":")
+
+def parse_options_and_blazon(query):
+	words = query.split()
+	options = list(itertools.takewhile(is_option_keyword, words))
+	blazon = " ".join(itertools.dropwhile(is_option_keyword, words))
+	return options, blazon
+
+
+async def heraldicon(session, query):
+	options, blazon = parse_options_and_blazon(query)
+	result = await utils.post_json(session, "https://2f1yb829vl.execute-api.eu-central-1.amazonaws.com/api",
+					{
+						"call": "generate-from-blazon",
+						"data": {
+							"blazon": blazon,
+							"options": options,
+						},
+					})
+	success = result.get("success")
+	if success:
+		png_url = success["png-url"]
+		edit_url = success['edit-url']
+		embed = embeds.DRAW.create("", edit_url, heading = "Success")
+		embed.set_thumbnail(url = "https://cdn.heraldicon.org/img/heraldicon-logo.png")
+		embed.set_image(url = png_url)
+		embed.add_field(name = "blazon", value = f"*{blazon}*", inline = True)
+		embed.set_footer(text = "CC BY-SA 4.0, attribution on https://heraldicon.org")
+		return embed
+
+	index = result["error"]["data"]["index"]
+	blazon = result["error"]["data"]["blazon"]
+	choices = result["error"]["data"]["auto-complete"]["choices"]
+
+	show_before = 20
+	show_after = 10
+	relevant_chunk = blazon[index - show_before:index]
+	bad_words = blazon[index:index + show_after]
+	if index > show_before:
+		relevant_chunk = "... " + relevant_chunk
+	if index + show_after < len(blazon):
+		bad_words += "..."
+	arrow = "-" * (1 + len(relevant_chunk)) + "^"
+	suggestions = [x[0] for x in choices[:20]]
+	error_message = f"""```Got confused by:
+'{relevant_chunk}{bad_words}'
+{arrow}
+
+Suggestions:
+{', '.join(suggestions)}
+	```"""
+	embed = embeds.DRAW.create("", error_message, heading = "Error")
+	embed.add_field(name = "blazon", value = f"*{blazon}*", inline = True)
+	embed.set_thumbnail(url = "https://cdn.heraldicon.org/img/heraldicon-logo.png")
+
+	return embed
+
+async def heraldicon_options(session):
+	result = await utils.post_json(session, "https://2f1yb829vl.execute-api.eu-central-1.amazonaws.com/api",
+					{
+						"call": "blazon-options",
+					})
+
```

Comment:
```suggestion
	result = await utils.post_json(
		session, 
		"https://2f1yb829vl.execute-api.eu-central-1.amazonaws.com/api",
		{"call": "blazon-options"}
	)
```
again, nicer formatting imo